src/java/map/FloatMask.java

```diff
@@ -761,6 +788,48 @@ public FloatMask spike(int radius, BinaryMask limiter) {
         return this;
     }
 
+    public FloatMask addFloatMaskCenteredAtLocationWithSize(FloatMask other, Vector2f location, int size) {
+        if (size > getSize()) {
+            throw new IllegalArgumentException("Added mask size is larger than base mask size");
+        }
+        FloatMask maskToBeAdded = other.copy().setSize(size);
+        add(maskToBeAdded, location, true);
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask useBrush(Vector2f location, String brushName, float intensity, int size) {
+        FloatMask brush = loadBrush(brushName, new SymmetrySettings(Symmetry.NONE, Symmetry.NONE, Symmetry.NONE));
+        brush.multiply(intensity / brush.getMax());
+        addFloatMaskCenteredAtLocationWithSize(brush, location, size);
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask useBrushRepeatedlyCenteredWithinArea(BinaryMask area, String brushName, int size, int frequency, float intensity) {
+        if (size > getSize()) {
+            throw new IllegalArgumentException("Added mask size is larger than base mask size");
+        }
+        List<Vector2f> possibleLocations = new ArrayList<Vector2f>(area.getAllCoordinatesEqualTo(true, 1));
+        int length = possibleLocations.size();
+        FloatMask brush = loadBrush(brushName, new SymmetrySettings(Symmetry.NONE, Symmetry.NONE, Symmetry.NONE));
+        brush.multiply(intensity / brush.getMax()).setSize(size);
+        for (int z = 0; z < frequency; z++) {
+            add(brush, possibleLocations.get(random.nextInt(length)), true);
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask useBrushRepeatedlyCenteredWithinAreaToDensity(BinaryMask area, String brushName, int size, float density, float intensity) {
+        List<Vector2f> possibleLocations = new ArrayList<Vector2f>(area.getAllCoordinatesEqualTo(true, 1));
+        int length = possibleLocations.size();
+        int frequency = (int) (density * (float) length / (float) 2621);
```

Comment:
what is with the arbitrary number? How does this then scale with mask size?

Reply:
Also you can add f to the end of a number to make it a float rather than casting it

Reply:
The number is approximately the area of 1 square km.  I thoroughly evaluated changing it with size and tested using that, and I determined this way seemed superior.
Changed to use f